evm/src/test/java/org/hyperledger/besu/evm/precompile/P256VerifyPrecompiledContractTest.java

```diff
@@ -87,4 +95,119 @@ void testP256Verify(final TestCase testCase) {
+
+    // Point in field but NOT on curve: these coordinates are valid field elements
+    // but do not satisfy the secp256r1 curve equation y² = x³ - 3x + b
+    String invalidPubKeyX = "79be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798";
+    String invalidPubKeyY = "b7c52588d95c3b9aa25b0403f1eef75702e84bb7597aabe663b82f6f04ef2777";
+
+    // Generate r,s values that could theoretically verify with this invalid point
+    // r component from a valid signature
+    String r = "c6047f9441ed7d6d3045406e95c07cd85c778e4b8cef3ca7abac09b95c709ee5";
+    // s component that would correspond to this scenario
+    String s = "30dae23890abb63e378e003d7f1d5006ab23cc7b3b65b3d0c7b45c7e1e2e08b9";
+
+    String input = messageHash + r + s + invalidPubKeyX + invalidPubKeyY;
+    Bytes inputBytes = Bytes.fromHexString(input);
+
+    // call default  implementation:
+    secp256R1.disableNative();
+    var pureJavaResult = contract.computeDefault(inputBytes);
+
+    // call maybeNative implementation
+    secp256R1.maybeEnableNative();
+    var maybeNativeResult = contract.computePrecompile(inputBytes, messageFrame);
+
+    // Should return empty bytes (INVALID) because curve validation catches
+    // the invalid point before signature verification attempts
+    assertTrue(
+        pureJavaResult.cachedResult().isSuccessful(),
+        "Invalid curve point should succeed with empty result");
+    assertEquals(
+        Bytes.EMPTY,
+        pureJavaResult.cachedResult().output(),
+        "Invalid curve point should return empty result");
+    assertEquals(
+        Bytes.EMPTY,
+        maybeNativeResult.output(),
+        "Invalid curve point should return empty result for native implementation");
+  }
+
+  @Test
+  void testModularComparisonWhenRPrimeExceedsN() {
+    // Test edge case where R'.x > n to verify proper modular comparison r' ≡ r (mod n)
+    // This tests that the implementation correctly handles cases where the computed
+    // point's x-coordinate exceeds the curve order
+    ECPoint G = P256VerifyPrecompiledContract.R1_PARAMS.getG(); // generator
+
+    // Create a scenario where we can control R' to have x-coordinate > n
+    // We'll use a known private key and construct a signature where R'.x > n but R'.x ≡ r (mod n)
+
+    // Message hash
+    String messageHash = "1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef";
+    BigInteger z = new BigInteger(messageHash, 16);
+
+    // Use a private key that will generate a point with large x-coordinate
+    BigInteger privateKey = N.subtract(BigInteger.valueOf(12345)); // close to n
+    ECPoint publicKeyPoint = G.multiply(privateKey).normalize();
+
+    // Create signature using a k value that produces R with x-coordinate > n
+    // We need R.x such that R.x > n but the verification still works via modular arithmetic
+    BigInteger k = N.add(BigInteger.valueOf(98765)); // This will create R.x > n
```

Comment:
I am not sure if this will create an R' whose x co-ordinate is greater than n. There is a spec test for this, so you could use the exact inputs (hash, r, s, x, y) to be sure: See https://github.com/ethereum/execution-spec-tests/blob/61f8ac90841770d9fc407f1498ec0e5229b27508/tests/osaka/eip7951_p256verify_precompiles/test_p256verify.py#L303-L315

Reply:
thanks using the values from execution-spec-tests makes for a cleaner test 👍